lib/adapter/mocha.js

```diff
@@ -21,12 +26,20 @@ function MochaReporter(runner, opts) {
 
   runner.on(EVENT_RUN_BEGIN, () => {
     client.createRun();
+
+    if (runner._workerReporter != undefined) {
+      //create tmp folder for all files
+      tmpDirPath = createTmpDir();
```

Comment:
I would suggest API of `artifacts` to be focused more on business logic

```
    if (runner._workerReporter != undefined) {
        startStoringArtifactsInFiles();
    }
```
Inside that function we create a temp dir and save a handler for it